lib/primitives/keyring.js

```diff
@@ -154,19 +155,25 @@ class KeyRing {
    * Inject data from public key.
    * @private
    * @param {Buffer} key
+   * @param {Boolean?} compress (default = true)
+   * @returns {KeyRing}
    */
 
-  fromPublic(key) {
+  fromPublic(key, compress) {
     assert(Buffer.isBuffer(key), 'Public key must be a buffer.');
     assert(secp256k1.publicKeyVerify(key), 'Not a valid public key.');
-    this.publicKey = key;
+    if (compress === false) {
+      this.publicKey = secp256k1.publicKeyConvert(key, compress === false);
```

Comment:
Could this be simplified?
```
if(!compress)
  this.publicKey = secp256k1.publicKeyConvert(key, false);
  ...
```
Or maybe closer resemblance to the rest of the module:
```
if(compress !== false)
  this.publicKey = secp256k1.publicKeyConvert(key, false);
  ...
```

Reply:
I changed this part a bit as this was problematic in some cases.